src/VirtualClient/VirtualClient.Dependencies/RedisPackageInstallation.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// in opensoruce to download redis from apt package manager. 
+namespace VirtualClient.Dependencies
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO.Abstractions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.VisualBasic;
+    using Newtonsoft.Json;
+    using Polly;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Contracts;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Platform;
+    using VirtualClient.Common.Telemetry;
+    using VirtualClient.Contracts;
+
+    /// <summary>
+    /// Provides functionality for installing latest version of Redis from Apt package manager on specific OS distribution version.
+    /// </summary>
+    [UnixCompatible]
+    public class RedisPackageInstallation : VirtualClientComponent
```

Comment:
Why do we need a RedisPackageInstallation to install redis from "apt" package stores? At a minimum, it seems like we should derive from LinuxPackageInstallation and simply override a few methods to keep the duplication to a minimum. This doesn't need to block this PR, but I'd like to think a bit more on propagating new dependency implementations before we figure out how to genericize the need. We might just need to extend LinuxPackageInstallation to have a few more features if it does not currently support the needs here.